### PR TITLE
Cross-compilation!

### PR DESCRIPTION
**Pull request by Mstrodl**
Working version of #4

### CODE REVIEW

**Comment by Mstrodl** [comment]
Blocking on ComputerScienceHouse/libgatekeeper#2

File: Cross.toml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[target.armv7-unknown-linux-gnueabihf]
+image = "nicholastmosher/gatekeeper:latest"
```
Comment (stevenmirabito) [inline Cross.toml]: This should probably be `computersciencehouse/gatekeeper-door:latest` now that it's being upstreamed, right?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+# Force rebuild when our versions change...
+RUN LIBNFC_VERSION=2b5ad9ce0be19fbca5abc04b4ee0b59fb612e590 && \
+    LIBFREEFARE_VERSION=e95406c0d1b417ff6db7ff8ee95df1b5981ec7b5 && \
+    LIBGATEKEEPER_VERSION=dc021430a68c66878bfe363fb85ce277e5a62501
+
+# We need to clone up here because installing armhf openssl breaks ca-certificates (seriously...)
+RUN git clone https://github.com/nfc-tools/libnfc && \
+    git clone https://github.com/nfc-tools/libfreefare && \
+    git clone https://github.com/Mstrodl/libgatekeeper
```
Comment (stevenmirabito) [inline Dockerfile]: TODO: need to update this to point to `ComputerScienceHouse/libgatekeeper` after https://github.com/ComputerScienceHouse/libgatekeeper/pull/2 is merged

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    mkdir build && \
+    cd build && \
+    cmake .. -DCMAKE_TOOLCHAIN_FILE=/armv7-toolchain.cmake \
+       -DCMAKE_INSTALL_PREFIX=/usr/arm-linux-gnueabihf/ \
+       -DBUILD_EXAMPLES=OFF \
+       -DBUILD_UTILS=OFF && \
+    make && \
+    make install
+
+#./configure --host=arm-linux-gnueabihf --prefix=/usr/arm-linux-gnueabihf/ && \
```
Comment (stevenmirabito) [inline Dockerfile]: No longer needed?

Comment (Mstrodl) [reply Dockerfile]: Yep sorry

File: Cross.toml (hunk 1 above)
Comment (Mstrodl) [reply Cross.toml]: Proooobably... Honestly wish docker would autobuild something if you put it in your PWD :/

File: Dockerfile (hunk 2 above)
Comment (Mstrodl) [reply Dockerfile]: This turned into a submodule so we're good :)

**Pull request merged**